runtime/compiler/aarch64/codegen/CallSnippet.cpp

```diff
@@ -0,0 +1,590 @@
+   return TR::InstOpCode::getOpCodeBinaryEncoding(TR::InstOpCode::bl) | ((distance >> 2) & 0x3ffffff); /* imm26 */
+   }
+
+TR_RuntimeHelper TR::ARM64CallSnippet::getHelper()
+   {
+   TR::Compilation * comp = cg()->comp();
+   TR_J9VMBase *fej9 = (TR_J9VMBase *) (cg()->fe());
+   TR::Node *callNode = getNode();
+   TR::SymbolReference *methodSymRef = callNode->getSymbolReference();
+   TR::MethodSymbol    *methodSymbol = methodSymRef->getSymbol()->castToMethodSymbol();
+   TR::SymbolReference *glueRef = NULL;
+
+   if (methodSymRef->isUnresolved() || comp->compileRelocatableCode())
+      {
+      if (methodSymbol->isSpecial())
+         return TR_ARM64interpreterUnresolvedSpecialGlue;
+      if (methodSymbol->isStatic())
+         return TR_ARM64interpreterUnresolvedStaticGlue;
+      return TR_ARM64interpreterUnresolvedDirectVirtualGlue;
+      }
+
+   if (methodSymbol->isVMInternalNative() || methodSymbol->isJITInternalNative())
+      return TR_ARM64nativeStaticHelper;
+
+   bool synchronised = methodSymbol->isSynchronised();
+
+   TR::DataType dataType = callNode->getDataType();
+   switch (dataType)
+      {
+      case TR::NoType:
+         return synchronised ? TR_ARM64interpreterSyncVoidStaticGlue : TR_ARM64interpreterVoidStaticGlue;
+      case TR::Int32:
+         return synchronised ? TR_ARM64interpreterSyncIntStaticGlue : TR_ARM64interpreterIntStaticGlue;
+      case TR::Int64:
+      case TR::Address:
+         return synchronised ? TR_ARM64interpreterSyncLongStaticGlue : TR_ARM64interpreterLongStaticGlue;
+      case TR::Float:
+         return synchronised ? TR_ARM64interpreterSyncFloatStaticGlue : TR_ARM64interpreterFloatStaticGlue;
+      case TR::Double:
+         return synchronised ? TR_ARM64interpreterSyncDoubleStaticGlue : TR_ARM64interpreterDoubleStaticGlue;
+      default:
+         TR_ASSERT(false, "Bad return data type '%s' for a call node.\n",
+                  cg()->getDebug()->getName(dataType));
+         return (TR_RuntimeHelper)0;
+      }
+   }
+
+uint8_t *TR::ARM64CallSnippet::emitSnippetBody()
+   {
+    uint8_t            *cursor = cg()->getBinaryBufferCursor();
+   TR::Node            *callNode = getNode();
+   TR::SymbolReference *methodSymRef = callNode->getSymbolReference();
+   TR::MethodSymbol    *methodSymbol = methodSymRef->getSymbol()->castToMethodSymbol();
+   TR::SymbolReference *glueRef;
+   TR::Compilation *comp = cg()->comp();
+   void               *trmpln = NULL;
+
+   getSnippetLabel()->setCodeLocation(cursor);
+
+   // Flush in-register arguments back to the stack for interpreter
+   cursor = flushArgumentsToStack(cursor, callNode, getSizeOfArguments(), cg());
+
+   glueRef = cg()->symRefTab()->findOrCreateRuntimeHelper(getHelper(), false, false, false);
+
+   // bl glueRef
+   *(int32_t *)cursor = encodeHelperBranchAndLink(glueRef, cursor, callNode, cg());
+   cursor += 4;
+
+   // Store the code cache RA
+   *(intptrj_t *)cursor = (intptrj_t)getCallRA();
```

Comment:
Nothing currently sets the RA in AArch64.  This is just a reminder in case you've forgotten about it.

Reply:
In the `p` version, `TR::PPCDepImmSymInstruction::generateBinaryEncoding()` calls `setCallRA()`.
I will revisit it when I work on that part.